### DISCUSSION TITLE
Output Glitch

### DISCUSSION THREAD

**Discussion by hittypitty** [joonaspaakko/Batch-Mockup-Smart-Object-Replacement-photoshop-script#8] [Q&A]
When I'm trying to run the batch on I say 10 input files, the batch runs fluently but the output is 8 out of 10 are glitched. That means the input images applied on the mockup are glitched. firstly I thought its an error with the ram, so I upgraded the ram from 8Gb to 16Gb but still, the error persists. Can you please help me with the same?

Attached some examples of the out I receive:
![GIRLS 5 - 2](https://user-images.githubusercontent.com/75974941/129488073-cb9def68-a5bc-47d8-a654-b4bbe01ac601.jpg)
![brother is my hero- both - 1](https://user-images.githubusercontent.com/75974941/129488082-c6efc067-09af-4ca0-9457-c42cbb41a49f.jpg)

**Comment by joonaspaakko** [comment]
I'm thinking this: [The last input file sticks around in the output when using noRepeats: true](https://github.com/joonaspaakko/Batch-Mockup-Smart-Object-Replacement-photoshop-script#5-the-last-input-file-sticks-around-in-the-output-when-using-norepeats-true). Well, technically not exactly the same scenario since you only have one Smart Object target here (so `noRepeats` setting doesn't matter), but without double checking, I'm pretty sure the script  forces the target layer to be visible, which could cause something like this to happen. Even if they have unique names it won't matter if the `SO`'s are duplicates. There might be something to improve there in the script. 

The first image I can't quite figure out, but the second looks more or less exactly what I would expect to happen in that scenario, aside from the top section that looks cut off...That's my best guess without seeing the mockup psd or the input files. 

**So you should perhaps try to:** 

1. Open the mockup `.psd`
2. Find the Smart Object layer with the **VISIBLE** placeholder graphic. 
    - Like perhaps with the `Move tool` and by right-clicking the graphic or by using the `Auto-select`.
    - If this **visible** placeholder graphic layer is not the one you were targeting before, then that was probably the issue.
3. Just to be on the safe side, give it a unique name 
4. Then use that layer as the target `SO` in the script settings.
    - Since that layer is already supposed to be the visible layer, it should get rid of the issue where the wrong layer is forcibly made visible.

**Reply by hittypitty** [reply]
These all things are already there as u mentioned. any other solution can be applied? @joonaspaakko

**Reply by joonaspaakko** [reply]
Is it possible the input files or the mockup `.psd` is corrupted? You should try to do the same process manually to see if anything changes.

**So to mimic what the script does:**

1. Open the mockup `.psd`.
2. Go edit the contents of the target `Smart Object`.
3. Hide any existing place holder graphics...
4. Add one of these input files `File > Place Embedded...`.
5. Save the `Smart Object` changes.
6. Save a copy of the mockup `.psd` in the same file format you used with the script.

If it still looks corrupted, then it's unrelated to the script, but if the issue disappears, that would confirm that it's the script that is causing it one way or another.

If this manually saved file looks fine, try the script again but try another output format to see if that changes anything:

```js
output: {
	format: 'jpg',     // 'jpg', 'png', 'tif', 'psd', 'pdf'
}
```

If these two tests don't help at all, I would need the mockup `.psd` and one of the problematic input files to test it myself.
